tests/testthat/test-expect-known.R

```diff
@@ -33,27 +33,23 @@ test_that("igores incomplete last line", {
 
 test_that("updates by default", {
   file <- tempfile()
-  write_lines("Hi!", file)
+  writeLines("Hi!", file)
   expect_failure(expect_known_output(cat("oops"), file, update = FALSE))
 
-  expect_equal(read_lines(file), "Hi!")
+  expect_equal(readLines(file), "Hi!")
   expect_failure(expect_known_output(cat("oops"), file, update = TRUE))
   expect_success(expect_known_output(cat("oops"), file))
 })
 
-test_that("works in non-UTF-8 locale", {
+test_that("works with utf-8 output", {
   text <- c("\u00fc", "\u2a5d", "\u6211", "\u0438")
-  file <- tempfile()
-  write_lines(text, file)
-
-  expect_success(expect_known_output(cat(text, sep = "\n"), file, update = FALSE))
-  withr::with_locale(
-    c(LC_CTYPE = "C"),
-    {
-      expect_false(l10n_info()$`UTF-8`)
-      expect_success(expect_known_output(cat(text, sep = "\n"), file, update = FALSE))
-    }
-  )
+
+  if (Sys.info()[["sysname"]] == "Windows") {
```

Comment:
@gaborcsardi does this seem like a reasonable approach for now?

Reply:
I think this is mostly fine for the test case. The `"\u00fc"` will (almost) always generate a warning on Windows, because that is transliterated into a latin1 or latin2 character that is cannot appear in UTF-8. But this is just the sign of a general problem.

In general we cannot currently assume that the reference file is in UTF-8. If want it in UTF-8, then we'll need to manually re-encode it, probably before writing it out with brio.